### PR TITLE
Map custom capability for viewing XB analytics

### PR DESCRIPTION
**Pull request by roborourke**
This maps a custom capability for viewing analytics to the `edit_posts` primitive which is required for anyone who can edit content.

### CODE REVIEW

**Comment by jazzsequence** [comment]
@roborourke 
1. What endpoint can/should I test to make sure I _don't_ have access if I am not logged in? (`/wp-json/analytics/v1/xbs/[...]/views?post_id=128&_locale=user`?)
2. Do I need to make changes in https://github.com/humanmade/aws-analytics/pull/153 to work with this change? (looks like not?)

**Comment by roborourke** [comment]
@jazzsequence 

1. It'll depend on the ID of the experience block but it's going to look something like `https://example.dev/wp-json/analytics/v1/xbs/<xb client id>/views`
2. Yeah, you'll need to match this pattern there as we don't want to use the read capability for audiences. Change the capability to something like `view_audience_estimates` and use the same `map_meta_cap` filter to map it to the `edit_posts` primitive

**Comment by jazzsequence** [comment]
Excellent, both those look good to me then 🚀🌔

**Review by jazzsequence — APPROVED** [review]
Looks good to me

File: inc/features/blocks/namespace.php [hunk 1]
```diff
@@ -186,7 +189,29 @@ function sanitize_id( $param ) : string {
  * @return boolean
  */
 function check_views_permission() : bool {
-	return current_user_can( 'edit_posts' );
+	return current_user_can( 'view_xb_analytics' );
```
Comment (rmccue) [inline inc/features/blocks/namespace.php:192]: Should this also be in the post type registration?

Comment (roborourke) [reply inc/features/blocks/namespace.php:192]: Why would it be there? This is a custom endpoint for reading analytics data for an XB, there's no post type (yet).

Comment (rmccue) [reply inc/features/blocks/namespace.php:192]: I was figuring that if you can read the audience, you'd be able to read the analytics data for it.

**Pull request merged**